eris_test.go

```diff
@@ -363,6 +363,128 @@ func TestErrorIs(t *testing.T) {
 	}
 }
 
+func TestErrorAs(t *testing.T) {
+	externalError := errors.New("external error")
+	rootErr := eris.New("root error")
+	wrappedErr := eris.Wrap(rootErr, "additional context")
+	customErr := withLayer{
+		msg: "additional context",
+		err: withEmptyLayer{
+			err: withMessage{
+				msg: "external error",
+			},
+		},
+	}
+
+	tests := map[string]struct {
+		cause  error       // root error
+		target interface{} // errors for comparison
+		match  bool        // expected comparison result
+		output interface{} // value of target on match
+	}{
+		"error against nil target": {
+			cause:  rootErr,
+			target: nil,
+			match:  false,
+			output: nil,
+		},
+		"error against non pointer interface": {
+			cause:  rootErr,
+			target: rootErr,
+			match:  false,
+			output: nil,
+		},
+		"error against non error interface": {
+			cause:  rootErr,
+			target: &withMessage{"test"},
+			match:  false,
+			output: nil,
+		},
+		"error against non pointer": {
+			cause:  rootErr,
+			target: "test",
+			match:  false,
+			output: nil,
+		},
+		"nil error against external target": {
+			cause:  nil,
+			target: &externalError,
+			match:  false,
+			output: nil,
+		},
+		"root error against external target": {
+			cause:  eris.New("root error"),
+			target: &externalError,
+			match:  false,
+			output: nil,
+		},
+		"wrapped error against external target": {
+			cause:  wrappedErr,
+			target: &externalError,
+			match:  false,
+			output: nil,
+		},
+		"nil wrapped error against root error target": {
+			cause:  eris.Wrap(nil, "additional context"),
+			target: &rootErr,
+			match:  false,
+			output: nil,
+		},
+		"wrapped error against its own root error target": {
+			cause:  wrappedErr,
+			target: &rootErr,
+			match:  true,
+			output: rootErr,
+		},
+		"root error against itself": {
+			cause:  rootErr,
+			target: &rootErr,
+			match:  true,
+			output: rootErr,
+		},
+		"root error against different root error": {
+			cause:  eris.New("other root error"),
+			target: &rootErr,
+			match:  false,
+			output: nil,
+		},
+		"wrapped error against same wrapped error": {
+			cause:  wrappedErr,
+			target: &wrappedErr,
+			match:  true,
+			output: wrappedErr,
+		},
+		"wrapped error against different wrapped error": {
+			cause:  eris.Wrap(nil, "some other error"),
+			target: &wrappedErr,
+			match:  false,
+			output: nil,
+		},
+		"custom error against similar type error": {
+			cause:  customErr,
+			target: &withLayer{msg: "additional layer", err: externalError},
+			match:  true,
```

Comment:
why does this case return true? seems like it should be false but i'm probably missing something

Reply:
I added custom As implementation for customError that's why, see the top of test file for that.

Reply:
hmm, i'm not seeing that in the diff though